crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -7720,7 +7722,10 @@ impl<'db> TypeInferenceBuilder<'db> {
             }
             KnownInstanceType::TypingSelf
             | KnownInstanceType::TypeAlias
+            | KnownInstanceType::TypedDict
```

Comment:
`TypedDict` might take parameters in the future: https://peps.python.org/pep-0764/

Reply:
The next release of `typing_extensions` will support subscripted inline `TypedDict`s (I merged support for it the other day): https://github.com/python/typing_extensions/pull/580

The PEP is still under discussion, but I'd say its chances are pretty high for being accepted in more-or-less its current form, and `typing_extensions` usually adds support for experimental features that are still under discussion so that users can play around with the proposed feature